Interactions, Slash-Commands, and Buttons, Stage Instance

nayvcake opened the pull request:
It is currently not ready yet. I'm studying the code before implementing it. You can use this branch to test a preview. If you find any problem in functions that have been implemented create a problem to solve a fix for this problem.


### Docs
This will not be documented, just read the comment that is in the Discord file or by the visual studio code (VSC/VSCODE).


### Development
- [x] Using API and Websocket in V9
- [x] Slash Command
- [x] Interactions
- [x] Stage Instance, Method (If there are methods implemented related to Stage Channel, comment that.)
- [x] Buttons
- [x] Example
- [x] **Interactions through web servers will not be possible to be implemented. That will be out of my focus. Maybe I'll change my mind and implement**
- [x] Permissions and Flags
- [x] Ephemeral messages
- [x] Creating commands (global/guild)
- [x] Editing commands (global/guild)
- [x] Commit/Update commands (Get the list of commands that have been set and update some information.)
- [x] Deleting commands (global/guild)
- [x] Add new events to this implementation focus
- [x] Thread
- [x] Selection Menu
- [x] Sticker
- [x] Sticker Events (Studying at the moment.)

nayvcake committed:
Merge pull request #2 from RabbitHouseCorp/master

Update

NavyCake commented:
There is an open PR related to Interactions.

NavyCake commented:
Plan changes. Returning with PR

nayvcake committed:
Merge pull request #4 from RabbitHouseCorp/master

Embeds

NavyCake commented:
Update list

nayvcake committed:
First Step to Thread

nayvcake committed:
Second step

nayvcake committed:
Merge pull request #6 from NavyCake/feature/thread

First Step to Thread

nayvcake committed:
Merge pull request #8 from NavyCake/feature/thread

Second step

NavyCake commented:
Latest thread implementations. It lacks some method to implement mainly in the Message class that I am still elaborating.

nayvcake committed:
Final part of the thread

nayvcake committed:
Finished

nayvcake committed:
Merge pull request #9 from NavyCake/feature/thread

Feature/thread

nayvcake committed:
Implementing Events

nayvcake committed:
Merge pull request #10 from NavyCake/feature/thread

Implementing Events

nayvcake committed:
Update ThreadChannel.js

nayvcake committed:
Merge pull request #11 from NavyCake/feature/thread

Update ThreadChannel.js

nayvcake committed:
Collect Thread mapping and send it to the channel map

nayvcake committed:
Merge pull request #12 from NavyCake/feature/thread

Collect Thread mapping and send it to the channel map

nayvcake committed:
Guilds get sticker support now

The implementation will be long as I don't have enough information to implement. I'm still studying how to add events to update guilds for new stickers.

NavyCake commented:
The implementation will be long as I don't have enough information to implement. I'm still studying how to add events to update guilds for new stickers. Soon the implementation will be released for later. As soon as I gather more information about Sticker. Code structure is strange.

nayvcake committed:
First message component implementation.

NavyCake commented:
I will create tools to capture interactions that are made. And also identify if the interaction has expired or still allow time to redeem it.

nayvcake committed:
fix

nayvcake committed:
url

NavyCake commented:
**Slash Command**
Example:
```js
bot.on('messageCreate', (message, slash) => {

})
```
When the event is not received from the server without interaction, it will return null, in which case it will return Slash

NavyCake commented:
Interaction response

**interaction.on('user', ...)**

You can limit the collector
**interaction.on('userIgnored', ...)**

nayvcake committed:
StickerPack

nayvcake committed:
new method

nayvcake committed:
More component implementations

nayvcake committed:
:tada: Interaction successfully implemented.

NavyCake commented:
I'm thinking of a good way to use Slash Command implementation, which will take a long time.

NavyCake commented:
I will add Task to be done

- [ ] Ephemeral messages (Bit **1 << 6**)
- [ ] MessageInteraction extends Message
- [ ] Test

nayvcake committed:
update

nayvcake committed:
update again

NavyCake commented:
Ephemeral will be defined by the Developer.

nayvcake committed:
Next commit will be the end of this implementation. To start Slash Command

Message interaction I will leave for tomorrow to finish.

DanielaGC committed:
Fix indent

NavyCake commented:
This PR is will be closed soon. Changes in this PR!

NavyCake commented:
I'm releasing a new library with compatibility for this Slash Command. And yes, with more extensive planning and giving possibilities for the developer to work with Slash Command with API Server and interactive for shards.
